src/main/java/sirius/db/mixing/types/MultiLanguageString.java

```diff
@@ -0,0 +1,174 @@
+     */
+    public String getRequiredText() {
+        return getRequiredText(defaultLanguage);
+    }
+
+    /**
+     * Returns the text associated with a given language.
+     *
+     * @param language the language code
+     * @return the text or an {@link sirius.kernel.health.HandledException} if not existent
+     */
+    public String getRequiredText(String language) {
+        if (!hasText(language)) {
+            throw Exceptions.createHandled()
+                            .withNLSKey("MultiLanguageString.textDoesNotExist")
+                            .set("language", language)
+                            .handle();
+        }
+        return fetchText(language);
+    }
+
+    /**
+     * Returns an optional text associated with the default language.
+     *
+     * @return an Optional String containing the text, otherwise an empty Optional
+     */
+    public Optional<String> getText() {
+        return getText(defaultLanguage);
+    }
+
+    /**
+     * Returns an optional text associated with a given language.
+     *
+     * @param language the language code
+     * @return an Optional String containing the text, otherwise an empty Optional
+     */
+    public Optional<String> getText(String language) {
+        if (!hasText(language)) {
+            return Optional.empty();
+        }
+        return Optional.of(fetchText(language));
+    }
+
+    /**
+     * Returns the text associated with the default language.
+     *
+     * @return the text if it exists, otherwise <tt>null</tt>
+     */
+    public String fetchText() {
+        return data().get(defaultLanguage);
+    }
+
+    /**
+     * Returns the text associated with a given language.
+     *
+     * @param language the language code
+     * @return the text if it exists, otherwise <tt>null</tt>
+     */
+    public String fetchText(String language) {
+        return data().get(language);
+    }
+
+    /**
+     * Returns the text associated with a given language, falling back to an alternative language when not found.
+     *
+     * @param language         the language code
+     * @param fallbackLanguage the alternative language code
+     * @return the text found under <tt>language</tt>, if none found under <tt>fallbackLanguage</tt>, <tt>null</tt> otherwise
+     */
+    public String fetchText(String language, String fallbackLanguage) {
+        return data().getOrDefault(language, data().getOrDefault(fallbackLanguage, fetchText(defaultLanguage)));
+    }
+
+    private void assertLanguage(String language) {
+        if (!supportedLanguages.contains(language)) {
+            throw Exceptions.createHandled()
+                            .withNLSKey("MultiLanguageString.invalidLanguage")
+                            .set("language", language)
+                            .handle();
+        }
+    }
+
+    public String getDefaultLanguage() {
+        return defaultLanguage;
+    }
+
+    public void setDefaultLanguage(String defaultLanguage) {
```

Comment:
is there a good reason to change this?

Reply:
It's an attempt to make the framework as generic as possible.
I'm thinking about a scenario where the default language could derive from the data context, like the assortment's default language, or a language picker int the UI, which could differ from the user's frontend display language.

Reply:
I'd personally make  the framework as simple as possible and stick with defaultLanguage == NLS.getCurrentLanguage - everything else should be handled one level higher(?) - IMHO

Reply:
Got rid of defaultLanguage. Using NLS.getCurrentLanguage instead.